### PR TITLE
feature/upgrade-to-symfony-3.4

### PR DESCRIPTION
**Pull request by jams246**

### CODE REVIEW

File: DependencyInjection/FlowsBuilderCompilerPass.php [hunk 1]
```diff
@@ -274,7 +274,7 @@ public function process(ContainerBuilder $container)
         $this->container = $container;
         $this->processorDefinitionsRegistry = $this->container->getDefinition('smartesb.registry.processor_definitions');
 
-        $processorDefinitionsServices = $container->findTaggedServiceIds(self::TAG_DEFINITIONS);
+        $processorDefinitionsServices = $container->findTaggedServiceIds(self::TAG_DEFINITIONS, null, null);
```
Comment (andreymor) [inline DependencyInjection/FlowsBuilderCompilerPass.php]: Why you are forcing null to a false parameter and adding a new one that does not exists?
SF 2.8
`public function findTaggedServiceIds($name)`
SF 3.4
`public function findTaggedServiceIds($name, $throwOnAbstract = false)`

File: Tests/DependencyInjection/FlowsBuilderCompilerPassTest.php [hunk 2]
```diff
@@ -95,12 +97,11 @@ public function testProcess()
             ->method('findTaggedServiceIds')
             ->willReturnMap(
                 [
-                    [FlowsBuilderCompilerPass::TAG_DEFINITIONS, ['serviceId' => [['nodeName' => 'abc']]]],
-                    ['producer.direct.demo', $serviceId],
-                    ['producer.custom.business_demo', $serviceId],
+                    [FlowsBuilderCompilerPass::TAG_DEFINITIONS, null, null, ['serviceId' => [['nodeName' => 'abc']]]],
```
Comment (andreymor) [inline Tests/DependencyInjection/FlowsBuilderCompilerPassTest.php]: Same here. Why nullable parameters to be mocked?

File: DependencyInjection/FlowsBuilderCompilerPass.php (hunk 1 above)
Comment (jams246) [reply DependencyInjection/FlowsBuilderCompilerPass.php]: If we're using SF 2.8 we do not need to map all parameters. IE: `$container->findTaggedServiceIds(self::TAG_DEFINITIONS);` will work as expected. But when upgrading to SF 3.4 we have to map all parameters.

I don't fully understand why SF3.4 requires it but based on the PHPUnit documentation https://phpunit.de/manual/6.5/en/test-doubles.html#test-doubles.stubs.examples.StubTest5.php, you can see that all 3 values must be mapped when using the `returnValueMap` feature.

File: Tests/DependencyInjection/FlowsBuilderCompilerPassTest.php (hunk 2 above)
Comment (jams246) [reply Tests/DependencyInjection/FlowsBuilderCompilerPassTest.php]: As explained in the other comment. SF2.8 does not require all map parameters but SF3.4 does.

**Review by andreymor — APPROVED** [review]

**Pull request merged**